src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/controller/ReferralController.kt

```diff
@@ -248,8 +248,12 @@ class ReferralController(
   }
 
   private fun getSentReferralForAuthenticatedUser(authentication: JwtAuthenticationToken, id: UUID): Referral {
-    val user = userMapper.fromToken(authentication)
-    return referralService.getSentReferralForUser(id, user)
+    return if (userMapper.isUserRequest(authentication)) {
+      val user = userMapper.fromToken(authentication)
+      referralService.getSentReferralForUser(id, user)
+    } else {
+      referralService.getSentReferralForRole(id, authentication.authorities)
```

Comment:
as mentioned above - should we validate the role outside the context of the referral and just provide a method to get the referral without any further checking?